### PR TITLE
statistics: use `[]float64` to represent hot peer loads

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?
Currently, all hot peer loads are saved in different variables and accessed using different functions.
Too many duplicated code and hard to extend.

### What is changed and how it works?
Save loads using `[]float64` and access them using enums.

### Check List

Tests
- Unit test

### Release note
- No release note

### CODE REVIEW

File: server/cluster/coordinator.go [hunk 1]
```diff
@@ -523,20 +523,21 @@ func (c *coordinator) collectHotSpotMetrics() {
 
 		stat, ok = status.AsLeader[storeID]
 		if ok {
-			hotSpotStatusGauge.WithLabelValues(storeAddress, storeLabel, "total_written_bytes_as_leader").Set(stat.TotalBytesRate)
-			hotSpotStatusGauge.WithLabelValues(storeAddress, storeLabel, "total_written_keys_as_leader").Set(stat.TotalKeysRate)
+			hotSpotStatusGauge.WithLabelValues(storeAddress, storeLabel, "total_written_bytes_as_leader").Set(stat.TotalLoads[statistics.RegionReadBytes])
```
Comment (rleungx) [inline server/cluster/coordinator.go:526]: write?

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Comment by lhy1024** [comment]
@Yisaer @rleungx  PTAL

**Review by Yisaer — APPROVED** [review]

File: server/statistics/hot_peer_cache.go [hunk 2]
```diff
@@ -396,18 +396,19 @@ func (f *hotPeerCache) updateHotPeerStat(newItem, oldItem *HotPeerStat, bytes, k
...
-		newItem.rollingByteRate.Add(bytes, interval)
-		newItem.rollingKeyRate.Add(keys, interval)
-		if newItem.rollingKeyRate.isFull() {
-			newItem.clearLastAverage()
+		newItem.rollingLoads = make([]*dimStat, len(regionStats))
+		for i, k := range regionStats {
+			ds := newDimStat(k)
+			ds.Add(deltaLoads[k], interval)
+			if ds.isFull() {
+				ds.clearLastAverage()
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go:404]: It will clear all.

File: server/statistics/hot_peer_cache.go [hunk 3]
```diff
@@ -419,10 +420,15 @@ func (f *hotPeerCache) updateHotPeerStat(newItem, oldItem *HotPeerStat, bytes, k
 	}
 
 	newItem.lastTransferLeaderTime = oldItem.lastTransferLeaderTime
-	newItem.rollingByteRate.Add(bytes, interval)
-	newItem.rollingKeyRate.Add(keys, interval)
+	var isFull bool
+	for i, k := range regionStats {
+		newItem.rollingLoads[i].Add(deltaLoads[k], interval)
+		if newItem.rollingLoads[i].isFull() {
+			isFull = true
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go:427]: break here?

**Review by rleungx — APPROVED** [review]

**Pull request merged**